lib/dependencies/ImportMetaPlugin.js

```diff
@@ -44,11 +45,40 @@ class ImportMetaPlugin {
 					return pathToFileURL(module.resource).toString();
 				};
 				/**
-				 * @param {Parser} parser parser
-				 * @param {Object} parserOptions parserOptions
+				 * @param {Parser} parser parser parser
+				 * @param {JavascriptParserOptions} parserOptions parserOptions
 				 * @returns {void}
 				 */
-				const parserHandler = (parser, parserOptions) => {
+				const parserHandler = (parser, { importMeta }) => {
+					if (importMeta === false) {
+						const { importMetaName } = compilation.outputOptions;
+						if (importMetaName === "import.meta") return;
+
+						parser.hooks.expression
+							.for("import.meta")
+							.tap("ImportMetaPlugin", metaProperty => {
+								const dep = new ConstDependency(
+									importMetaName,
+									metaProperty.range
+								);
+								dep.loc = metaProperty.loc;
+								parser.state.module.addPresentationalDependency(dep);
+								return true;
+							});
+						parser.hooks.unhandledExpressionMemberChain
+							.for("import.meta")
+							.tap("ImportMetaPlugin", (expr, members) => {
+								const dep = new ConstDependency(
+									`${importMetaName}${propertyAccess(members, 0)}`,
+									expr.range
+								);
+								dep.loc = expr.loc;
+								parser.state.module.addPresentationalDependency(dep);
+								return true;
+							});
```

Comment:
```suggestion
```

That's not really needed since it will fallback to `hooks.expression` in the end anyway.

Reply:
technically this is faster.. anyway I have removed this `tap`